pkgs/tools/filesystems/unionfs-fuse/default.nix

```
@@ -12,23 +12,31 @@ stdenv.mkDerivation rec {
   };
 
   patches =
-    [ # Prevent the unionfs daemon from being killed during
+    [
+      # Prevent the unionfs daemon from being killed during
       # shutdown. See
       # http://www.freedesktop.org/wiki/Software/systemd/RootStorageDaemons/
       # for details.
       ./prevent-kill-on-shutdown.patch
     ];
 
+  postPatch = lib.optionals stdenv.isDarwin ''
+    substituteInPlace CMakeLists.txt \
+      --replace '/usr/local/include/osxfuse/fuse' '${osxfuse}/include/osxfuse/fuse'
+    substituteInPlace src/CMakeLists.txt \
+      --replace 'target_link_libraries(unionfs fuse pthread)' 'target_link_libraries(unionfs osxfuse pthread)'
+  '';
+
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ fuse ];
+  buildInputs = lib.optional stdenv.isLinux fuse ++ lib.optional stdenv.isDarwin osxfuse;
```

I think everyone else uses fuse:

```suggestion
  buildInputs = [ (if stdenv.isDarwin osxfuse else fuse) ];
```